src/main/scala/io/iohk/ethereum/blockchain/sync/SyncController.scala

```diff
@@ -198,7 +198,7 @@ class SyncController(
       else
         (SyncState.empty, fastSyncStateStorage.purge())
 
-    private var mptNodesQueue: Set[HashType] = initialSyncState.mptNodesQueue
+    private var mptNodesQueue: Seq[HashType] = initialSyncState.mptNodesQueue
     private var nonMptNodesQueue: Set[HashType] = initialSyncState.nonMptNodesQueue
     private var blockBodiesQueue: Set[ByteString] = initialSyncState.blockBodiesQueue
     private var receiptsQueue: Set[ByteString] = initialSyncState.receiptsQueue
```

Comment:
> order is not required

It's not about whether we need order or not. Prepending to a `List` is much faster than adding to a `Set`, especially when we're talking about sizeable `ByteStrings`!